demo/python/astronomy.py

```diff
@@ -551,18 +551,18 @@ def AngleBetween(a, b):
         The angle between the two vectors expressed in degrees.
         The value is in the range [0, 180].
     """
-    r = a.Length() * b.Length()
+    r: float = a.Length() * b.Length()
     if r < 1.0e-8:
-        return BadVectorError()
-    dot = (a.x*b.x + a.y*b.y + a.z*b.z) / r
+        raise BadVectorError()
```

Comment:
Wow, that was a serious bug! I'm surprised pylint didn't complain about that.

Reply:
> Wow, that was a serious bug! I'm surprised pylint didn't complain about that.
Yup, I didn't notice it at first either but mypy immediately flagged it, working out already haha!